perun-core/src/main/java/cz/metacentrum/perun/core/api/ResourcesManager.java

```diff
@@ -65,6 +65,21 @@
 	 */
 	Resource createResource(PerunSession perunSession, Resource resource, Vo vo, Facility facility) throws InternalErrorException, PrivilegeException, VoNotExistsException, FacilityNotExistsException, ResourceExistsException;
 
+	/**
+	 * Copy "template" settings(attributes,services,tags (if exists),...) from user's another existing resource and create new resource with this template.
+	 * Resource can be from any of user's facilities.
+	 *
+	 * @param perunSession
+	 * @param templateResource template resource to copy
+	 * @param destinationResource destination resource containing destination facility, VO & resource name.
+	 * @param withGroups if set to true and resources ARE from the same VO we also copy all group-resource and member-resource attributes,
+	 *                   if set to true and resources ARE NOT from the same VO we copy resource only attributes,
+	 *                   if set to false we copy resource only attributes.
```

Comment:
If set to true and resources ARE from the same VO => we will copy not only attributes but also assign all groups same as on the templateResource.
If set to true and resources ARE NOT from the same VO => I would prefer exception to be thrown, because this should be forbidden state.
If set to false => We will not copy groups and group-related attributes. There are still tags, services etc.